### ISSUE TITLE
Fixing running silentbridge in Kali Linux 2024

### ISSUE THREAD

**Issue by Ab4y98** [s0lst1c3/silentbridge#9]
Hi,

I’m opening this issue because after using the tool SilentBridge during a recent engagement, I encountered some dependency issues when running it with Python 3. It’s possible these issues stem from the tool being around 6 years old and possibly originally written for Python 2.7. Regardless, I’ve written a small script to address some of the headaches I faced with the tool, and I’d like to share it in case it’s helpful to others.

After updating Kali and downloading the tool we can see the following error:

![image](https://github.com/user-attachments/assets/a581b6bd-6766-4adb-9efd-57a8727630a4)

And I've tries so many approaches to fix it.. and came up with this solution:

`wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
sudo python2.7 get-pip.py
pip install virtualenv==20.21.1 --break-system-packages --ignore-installed
virtualenv -p $(which python2) venv2
source venv2/bin/activate
git clone https://github.com/s0lst1c3/silentbridge.git
pip install scapy==2.4.3 --ignore-installed
pip install netifaces
pip install nanpy
cd silentbridge
python2 ./silentbridge`

I'll let ChatGPT do the rest of the talking :)

Here's a brief explanation of the commands:

1. **Install `pip` for Python 2.7**:
   ```bash
   wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
   sudo python2.7 get-pip.py
   ```
   Downloads and installs `pip` for Python 2.7.

2. **Install specific version of `virtualenv`**:
   ```bash
   pip install virtualenv==20.21.1 --break-system-packages --ignore-installed
   ```
   Installs `virtualenv` version 20.21.1 while bypassing system package conflicts.

3. **Create a virtual environment with Python 2.7**:
   ```bash
   virtualenv -p $(which python2) venv2
   source venv2/bin/activate
   ```
   Creates and activates a virtual environment using Python 2.7.

4. **Clone SilentBridge repository and install dependencies**:
   ```bash
   git clone https://github.com/s0lst1c3/silentbridge.git
   pip install scapy==2.4.3 --ignore-installed
   pip install netifaces
   pip install nanpy
   ```
   Clones the SilentBridge repository and installs required dependencies (`scapy`, `netifaces`, and `nanpy`).

5. **Run SilentBridge**:
   ```bash
   cd silentbridge
   python2 ./silentbridge
   ```
   Navigates to the SilentBridge directory and runs the tool with Python 2.7.

---

In short, these commands set up a Python 2.7 environment, install dependencies, and run the SilentBridge tool.

![image](https://github.com/user-attachments/assets/963a33dc-0a6c-4dfa-a572-42fc3a9faec1)

**Comment by joshuanutt** [comment]
Ran into this error when installing netifaces:
```bash
netifaces.c:1:10: fatal error: Python.h: No such file or directory
      1 | #include <Python.h>
        |          ^~~~~~~~~~
  compilation terminated.
```

Fixed with:
`sudo apt-get install python2-dev`

After that, this worked fine on Kali

**Comment by l4rm4nd** [comment]
Combined into one bash script:

````bash
# install dependencies for netifaces
sudo apt install python2-dev -y

# get python2 pip
wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
sudo python2.7 get-pip.py
rm get-pip.py

# downgrade virtualenv to support python2.7 venvs
# see https://github.com/pypa/virtualenv/releases/tag/20.22.0 - drops python2.7 support
pip install virtualenv==20.21.1 --break-system-packages --ignore-installed

# clone silentbridge
git clone https://github.com/s0lst1c3/silentbridge && cd silentbridge

# create venv for python2.7
virtualenv -p $(which python2) venv2
source venv2/bin/activate

# install silentbridge dependencies
pip install scapy==2.4.3 --ignore-installed
pip install netifaces
pip install nanpy

# test run silentbridge
python2 ./silentbridge
````

Works on Raspberry Pi 4 with `kali-linux-2025.2-raspberry-pi-arm64.img`.
